x-pack/plugins/security_solution/public/app/app.tsx

```diff
@@ -33,6 +34,7 @@ interface StartAppComponent {
   onAppLeave: (handler: AppLeaveHandler) => void;
   setHeaderActionMenu: AppMountParameters['setHeaderActionMenu'];
   store: Store<State, Action>;
+  theme$: AppMountParameters['theme$'];
```

Comment:
I noticed that in the other places we grab `theme$` from the `useKibana` hook that's also used in this file. Just curious is there a reason to use one over the other - the passed in param vs. from the hook?

Reply:
This is a good question. There isn't much of a reason other than I felt the decision to use it in some places and not in others just was a question of clarity. Here in the start component I felt it was clearer to include it as a prop rather than pulling it out of the services.

So rather than pulling it out of here from the useKibana hook:

https://github.com/elastic/kibana/blob/c70178d54295f67ed6fa009ede67c53a85e66ae0/x-pack/plugins/security_solution/public/app/app.tsx#L47-L50

I felt it was clearer to show where exactly this `theme$` property is coming from which is on app mount, passed in here:

https://github.com/elastic/kibana/blob/c70178d54295f67ed6fa009ede67c53a85e66ae0/x-pack/plugins/security_solution/public/plugin.tsx#L144-L158

And then obviously rather than pass it down between nested components to just use the `useKibana` hook made more sense when deep into the application.

But I felt at the higher level where we are starting the app it just looked a little clearer than pulling it from the hook as it shows where in the types this is coming from.

Reply:
Thanks for the reply! Makes sense.